File: app/src/main/java/com/callscribe/data/Languages.kt
package com.callscribe.data

/**
 * Езикът важи и за двете стъпки: подава се на сървъра за транскрипция и определя
 * на какъв език моделът пише извлечените задачи.
 */
object Languages {

    data class Option(val code: String, val label: String)

    val all = listOf(
        Option("bg", "Български"),
        Option("en", "English")
    )

    const val DEFAULT = "bg"

    fun label(code: String): String =
        all.firstOrNull { it.code == code }?.label ?: code

    fun normalize(code: String): String =
        if (all.any { it.code == code }) code else DEFAULT
}

File: ios/App/Sources/Models/Languages.swift
import Foundation

/// Езикът важи и за двете стъпки: подава се на сървъра за транскрипция и определя
/// на какъв език моделът пише извлечените задачи.
enum Languages {

    struct Option: Identifiable, Hashable {
        let code: String
        let label: String
        var id: String { code }
    }

    static let all: [Option] = [
        Option(code: "bg", label: "Български"),
        Option(code: "en", label: "English")
    ]

    static let defaultCode = "bg"

    static func label(_ code: String) -> String {
        all.first { $0.code == code }?.label ?? code
    }

    static func normalize(_ code: String) -> String {
        all.contains { $0.code == code } ? code : defaultCode
    }
}
